src/adapter/4C_adapter_str_structure_new.cpp

```diff
@@ -187,9 +185,27 @@ void Adapter::StructureBaseAlgorithmNew::setup_tim_int()
         return Core::Binstrategy::DefaultRelevantPoints{}(discret, ele, disnp);
     };
 
-    Core::Rebalance::rebalance_discretizations_by_binning(binning_params,
-        Global::Problem::instance()->output_control_file(), actdis_vec, correct_node,
-        determine_relevant_points, true);
+    int node_distribution_repair_necessary = 0;
+    try
+    {
+      Core::Rebalance::rebalance_discretizations_by_binning(binning_params,
+          Global::Problem::instance()->output_control_file(), actdis_vec, correct_node,
+          determine_relevant_points, true);
+    }
+    catch (const Core::Exception& e)
+    {
+      // this can happen due to improper node distribution in assign degrees of freedom
+      node_distribution_repair_necessary = 1;
+    }
```

Comment:
Hm, 4C isn't really exception-safe, so doing try-catch on a large algorithmic component seems to beg for trouble. Can you guarantee that the objects are in a useable/defined state if an exception is thrown somewhere in the middle? If that was always in this PR I have missed it, sorry for the late notice in this case.